### ISSUE TITLE
External management of Trust

### ISSUE THREAD

**Issue by vincentparrett** [OctopusDeploy/Halibut#70]
I'm investigating adding support for the MS Generic hosting environment, and have gotten quite far. One issue I have found though, is that the abstraction hides access to the runtime, which makes adding and removing trusted thumbprints difficult. 

What are your thoughts on adding a ITrustProvider (and a default implementation to use when not provided), and a constructor on HalibutRuntime to take one in.

Something like this :

````c#
public interface ITrustProvider 
{
  void Add(string clientThumbprint);
  void Remove(string clientThumbprint);
  bool IsTrusted(string clientThumbprint);
}

public class HalibutRuntime : IHalibutRuntime
{
        public HalibutRuntime(IServiceFactory serviceFactory, X509Certificate2 serverCertificate, ITrustProvider trustProvider)

}

````
This would allow us to inject the ITrustProvider elsewhere in our application to manage the trust without needing access to the Halibut runtime (which in my case would be contained in an IHostedService instance (which would then take the ITrustProvider in it's constructor).

**Comment by vincentparrett** [comment]
PR submitted, was a really simple change.

**Comment by vincentparrett** [comment]
Closing, implemented in PR #81
